src/main/kotlin/org/jmailen/gradle/kotlinter/support/EditorConfigUtils.kt

```diff
@@ -12,20 +15,39 @@ internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverr
     val rules = ktLintParams.disabledRules
 
     return if (rules.isEmpty()) {
-        EditorConfigOverride.emptyEditorConfigOverride
+        EMPTY_EDITOR_CONFIG_OVERRIDE
     } else {
-        EditorConfigOverride.from(DefaultEditorConfigProperties.ktlintDisabledRulesProperty to rules.joinToString(separator = ","))
+        rules
+            .asSequence()
+            .map(::getDisabledRulePropertyName)
+            .map { propertyName ->
+                EditorConfigProperty(
+                    type = PropertyType(propertyName, "Rule to be disabled", IDENTITY_VALUE_PARSER),
+                    defaultValue = "disabled",
+                )
+            }
+            .map { it to "disabled" }
+            .toList()
+            .toTypedArray()
+            .let { EditorConfigOverride.from(*it) }
     }
 }
 
-internal fun resetEditorconfigCacheIfNeeded(
+private fun getDisabledRulePropertyName(it: String) =
+    if (it.contains(':')) { // Rule from a custom source set
```

Comment:
⛏️ 
```suggestion
    if (it.contains(':')) { // Rule from a custom rule set
```

📝 
I'll comment out this also includes rules from experimental rule set so maybe:
```suggestion
    if (it.contains(':')) { // Rule from a non-standard ruleset
```